{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bigrams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import string\n",
    "import nltk\n",
    "from nltk.corpus import stopwords\n",
    "import seaborn as sns\n",
    "from nltk.tokenize import word_tokenize \n",
    "from nltk.collocations import *\n",
    "import nltk\n",
    "from nltk.stem import WordNetLemmatizer\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Journal</th>\n",
       "      <th>Title</th>\n",
       "      <th>Authors</th>\n",
       "      <th>Year</th>\n",
       "      <th>Abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>International journal of systematic and evolut...</td>\n",
       "      <td>Leucothrix arctica sp. nov., isolated from Arc...</td>\n",
       "      <td>['Baek Kiwoon' 'Choi Ahyoung' 'Lee Yung Mi' 'L...</td>\n",
       "      <td>2018</td>\n",
       "      <td>A Gram-stain-negative, non-motile, oxidase- an...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>International journal of systematic and evolut...</td>\n",
       "      <td>Hyphobacterium indicum sp. nov., isolated from...</td>\n",
       "      <td>['Ruan Chu-Jin' 'Zheng Xiao-Wei' 'Wang Jian' '...</td>\n",
       "      <td>2018</td>\n",
       "      <td>A novel aerobic, Gram-stain-negative bacterium...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>International journal of systematic and evolut...</td>\n",
       "      <td>Sphingopyxis lindanitolerans sp. nov. strain W...</td>\n",
       "      <td>['Kaminski Michal A' 'Sobczak Adam' 'Spolnik G...</td>\n",
       "      <td>2018</td>\n",
       "      <td>An aerobic, Gram-stain-negative, rod-shaped, n...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>International journal of systematic and evolut...</td>\n",
       "      <td>Mesosutterella multiformis gen. nov., sp. nov....</td>\n",
       "      <td>['Sakamoto Mitsuo' 'Ikeyama Nao' 'Kunihiro Tad...</td>\n",
       "      <td>2018</td>\n",
       "      <td>Two novel, obligately anaerobic, Gram-stain-ne...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>International journal of systematic and evolut...</td>\n",
       "      <td>Acinetobacter sichuanensis sp. nov., recovered...</td>\n",
       "      <td>['Qin Jiayuan' 'Hu Yiyi' 'Feng Yu' 'Lv Xiaoju'...</td>\n",
       "      <td>2018</td>\n",
       "      <td>A novel Acinetobacter strain, WCHAc060041(T), ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Journal  \\\n",
       "0  International journal of systematic and evolut...   \n",
       "1  International journal of systematic and evolut...   \n",
       "4  International journal of systematic and evolut...   \n",
       "5  International journal of systematic and evolut...   \n",
       "6  International journal of systematic and evolut...   \n",
       "\n",
       "                                               Title  \\\n",
       "0  Leucothrix arctica sp. nov., isolated from Arc...   \n",
       "1  Hyphobacterium indicum sp. nov., isolated from...   \n",
       "4  Sphingopyxis lindanitolerans sp. nov. strain W...   \n",
       "5  Mesosutterella multiformis gen. nov., sp. nov....   \n",
       "6  Acinetobacter sichuanensis sp. nov., recovered...   \n",
       "\n",
       "                                             Authors  Year  \\\n",
       "0  ['Baek Kiwoon' 'Choi Ahyoung' 'Lee Yung Mi' 'L...  2018   \n",
       "1  ['Ruan Chu-Jin' 'Zheng Xiao-Wei' 'Wang Jian' '...  2018   \n",
       "4  ['Kaminski Michal A' 'Sobczak Adam' 'Spolnik G...  2018   \n",
       "5  ['Sakamoto Mitsuo' 'Ikeyama Nao' 'Kunihiro Tad...  2018   \n",
       "6  ['Qin Jiayuan' 'Hu Yiyi' 'Feng Yu' 'Lv Xiaoju'...  2018   \n",
       "\n",
       "                                            Abstract  \n",
       "0  A Gram-stain-negative, non-motile, oxidase- an...  \n",
       "1  A novel aerobic, Gram-stain-negative bacterium...  \n",
       "4  An aerobic, Gram-stain-negative, rod-shaped, n...  \n",
       "5  Two novel, obligately anaerobic, Gram-stain-ne...  \n",
       "6  A novel Acinetobacter strain, WCHAc060041(T), ...  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "papers = pd.read_csv(\"tabela_1_ed_sem_2019.csv\", index_col=[0]) #dataframe derived from medline file\n",
    "papers.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "abstracts_full = papers[[\"Abstract\", \"Year\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "stop = stopwords.words('english')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def freq_bigram_by_year(year):\n",
    "    abstracts_year = abstracts_full[abstracts_full[\"Year\"] == year][\"Abstract\"]\n",
    "    \n",
    "    abstracts_year_cat = abstracts_year.str.cat(sep=\" \")\n",
    "    \n",
    "    punct = '!\"#$%&\\'()*+,./:;<=>?@[\\\\]^_`{}~'   # `|` is not present here\n",
    "    transtab = str.maketrans(dict.fromkeys(punct, ''))\n",
    "    abstracts_year_no_punct = abstracts_year_cat.translate(transtab)\n",
    "    \n",
    "    stop_words = stopwords.words('english')\n",
    "    word_tokens_year = word_tokenize(abstracts_year_no_punct.lower()) \n",
    "    \n",
    "    filtered_sentence_year = [] \n",
    "    for w in word_tokens_year: \n",
    "        if w not in stop_words: \n",
    "            filtered_sentence_year.append(w)\n",
    "    \n",
    "    wordnet_lemmatizer = WordNetLemmatizer()\n",
    "    lemmatized_year = []\n",
    "    \n",
    "    \n",
    "    for w in filtered_sentence_year:\n",
    "        lemmatized_w = wordnet_lemmatizer.lemmatize(w)\n",
    "        lemmatized_year.append(lemmatized_w)\n",
    "    \n",
    "    #bigrams\n",
    "    bigramFinder_year = nltk.collocations.BigramCollocationFinder.from_words(lemmatized_year)\n",
    "    bigram_freq_year = bigramFinder_year.ngram_fd.items()\n",
    "    bigramFreqTable_year = pd.DataFrame(list(bigram_freq_year), columns=['bigram','freq']).sort_values(by='freq', ascending=False)\n",
    "    bigramFreqTable_year.to_csv(\"{}_bigram_freq.csv\".format(year))\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(2001,2019):\n",
    "    freq_bigram_by_year(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bigram</th>\n",
       "      <th>freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>(type, strain)</td>\n",
       "      <td>13821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>(16, rrna)</td>\n",
       "      <td>13082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>(rrna, gene)</td>\n",
       "      <td>12456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>(gene, sequence)</td>\n",
       "      <td>11829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>(sp, nov)</td>\n",
       "      <td>10471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>(fatty, acid)</td>\n",
       "      <td>8252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>(novel, specie)</td>\n",
       "      <td>8084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>(specie, genus)</td>\n",
       "      <td>6813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>(sequence, similarity)</td>\n",
       "      <td>6794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>(nov, proposed)</td>\n",
       "      <td>6578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>(degree, c)</td>\n",
       "      <td>6487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>(gc, content)</td>\n",
       "      <td>6397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>(proposed, type)</td>\n",
       "      <td>5644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>(phylogenetic, analysis)</td>\n",
       "      <td>5275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>(dna, gc)</td>\n",
       "      <td>4642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>(polar, lipid)</td>\n",
       "      <td>4207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>(based, 16)</td>\n",
       "      <td>4007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>(closely, related)</td>\n",
       "      <td>3723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>(represents, novel)</td>\n",
       "      <td>3287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>(analysis, based)</td>\n",
       "      <td>3163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>(major, fatty)</td>\n",
       "      <td>3112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>(dna-dna, hybridization)</td>\n",
       "      <td>3010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>(sequence, analysis)</td>\n",
       "      <td>2715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>678</th>\n",
       "      <td>(member, genus)</td>\n",
       "      <td>2498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>(c, ph)</td>\n",
       "      <td>2497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1298</th>\n",
       "      <td>(nov, type)</td>\n",
       "      <td>2440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>942</th>\n",
       "      <td>(dna-dna, relatedness)</td>\n",
       "      <td>2434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>620</th>\n",
       "      <td>(cellular, fatty)</td>\n",
       "      <td>2371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>(summed, feature)</td>\n",
       "      <td>2224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1520</th>\n",
       "      <td>(iso-c15, 0)</td>\n",
       "      <td>2173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170657</th>\n",
       "      <td>(nnt, 9687)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170670</th>\n",
       "      <td>(g, xinjiangensis)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170654</th>\n",
       "      <td>(tp2-8t, 9706)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170652</th>\n",
       "      <td>(similarity, 9706)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170651</th>\n",
       "      <td>(j2, similarity)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170649</th>\n",
       "      <td>(relative, gracilibacillus)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170648</th>\n",
       "      <td>(gracilibacillus, closest)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170647</th>\n",
       "      <td>(sg103t, member)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170646</th>\n",
       "      <td>(suggested, sg103t)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170669</th>\n",
       "      <td>(9687, 16)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170671</th>\n",
       "      <td>(j2, g)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170692</th>\n",
       "      <td>(plant-growth-promoting, ability)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170684</th>\n",
       "      <td>(sg103t, jcm)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170691</th>\n",
       "      <td>(evaluation, plant-growth-promoting)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170690</th>\n",
       "      <td>(winteri, natural)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170689</th>\n",
       "      <td>(phosphate-mobilizing, gram-negative)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170688</th>\n",
       "      <td>(14683t, phosphate-mobilizing)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170687</th>\n",
       "      <td>(ncimb, 14683t)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170686</th>\n",
       "      <td>(17253t, ncimb)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170685</th>\n",
       "      <td>(jcm, 17253t)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170683</th>\n",
       "      <td>(alcaliphilus, sp)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170673</th>\n",
       "      <td>(tp2-8t, g)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170682</th>\n",
       "      <td>(gracilibacillus, alcaliphilus)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170681</th>\n",
       "      <td>(sg103t, merit)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170679</th>\n",
       "      <td>(3-2, relatedness)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170678</th>\n",
       "      <td>(4-9, 3-2)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170677</th>\n",
       "      <td>(2-4, 4-9)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170676</th>\n",
       "      <td>(revealed, 2-4)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170675</th>\n",
       "      <td>(nnt, revealed)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408879</th>\n",
       "      <td>(antitubercular, therapy)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>408880 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       bigram   freq\n",
       "28                             (type, strain)  13821\n",
       "16                                 (16, rrna)  13082\n",
       "17                               (rrna, gene)  12456\n",
       "18                           (gene, sequence)  11829\n",
       "129                                 (sp, nov)  10471\n",
       "78                              (fatty, acid)   8252\n",
       "123                           (novel, specie)   8084\n",
       "124                           (specie, genus)   6813\n",
       "40                     (sequence, similarity)   6794\n",
       "130                           (nov, proposed)   6578\n",
       "69                                (degree, c)   6487\n",
       "59                              (gc, content)   6397\n",
       "131                          (proposed, type)   5644\n",
       "185                  (phylogenetic, analysis)   5275\n",
       "58                                  (dna, gc)   4642\n",
       "104                            (polar, lipid)   4207\n",
       "187                               (based, 16)   4007\n",
       "26                         (closely, related)   3723\n",
       "122                       (represents, novel)   3287\n",
       "186                         (analysis, based)   3163\n",
       "77                             (major, fatty)   3112\n",
       "340                  (dna-dna, hybridization)   3010\n",
       "19                       (sequence, analysis)   2715\n",
       "678                           (member, genus)   2498\n",
       "70                                    (c, ph)   2497\n",
       "1298                              (nov, type)   2440\n",
       "942                    (dna-dna, relatedness)   2434\n",
       "620                         (cellular, fatty)   2371\n",
       "80                          (summed, feature)   2224\n",
       "1520                             (iso-c15, 0)   2173\n",
       "...                                       ...    ...\n",
       "170657                            (nnt, 9687)      1\n",
       "170670                     (g, xinjiangensis)      1\n",
       "170654                         (tp2-8t, 9706)      1\n",
       "170652                     (similarity, 9706)      1\n",
       "170651                       (j2, similarity)      1\n",
       "170649            (relative, gracilibacillus)      1\n",
       "170648             (gracilibacillus, closest)      1\n",
       "170647                       (sg103t, member)      1\n",
       "170646                    (suggested, sg103t)      1\n",
       "170669                             (9687, 16)      1\n",
       "170671                                (j2, g)      1\n",
       "170692      (plant-growth-promoting, ability)      1\n",
       "170684                          (sg103t, jcm)      1\n",
       "170691   (evaluation, plant-growth-promoting)      1\n",
       "170690                     (winteri, natural)      1\n",
       "170689  (phosphate-mobilizing, gram-negative)      1\n",
       "170688         (14683t, phosphate-mobilizing)      1\n",
       "170687                        (ncimb, 14683t)      1\n",
       "170686                        (17253t, ncimb)      1\n",
       "170685                          (jcm, 17253t)      1\n",
       "170683                     (alcaliphilus, sp)      1\n",
       "170673                            (tp2-8t, g)      1\n",
       "170682        (gracilibacillus, alcaliphilus)      1\n",
       "170681                        (sg103t, merit)      1\n",
       "170679                     (3-2, relatedness)      1\n",
       "170678                             (4-9, 3-2)      1\n",
       "170677                             (2-4, 4-9)      1\n",
       "170676                        (revealed, 2-4)      1\n",
       "170675                        (nnt, revealed)      1\n",
       "408879              (antitubercular, therapy)      1\n",
       "\n",
       "[408880 rows x 2 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#frequency from all\n",
    "abstracts_all = abstracts_full[\"Abstract\"]\n",
    "abstracts_cat = abstracts_all.str.cat(sep=\" \")\n",
    "\n",
    "punct = '!\"#$%&\\'()*+,./:;<=>?@[\\\\]^_`{}~'   # `|` is not present here\n",
    "transtab = str.maketrans(dict.fromkeys(punct, ''))\n",
    "abstracts_all_no_punct = abstracts_cat.translate(transtab)\n",
    "\n",
    "stop_words = stopwords.words('english')\n",
    "word_tokens = word_tokenize(abstracts_all_no_punct.lower()) \n",
    "  \n",
    "filtered_sentence_all = [] \n",
    "  \n",
    "for w in word_tokens: \n",
    "    if w not in stop_words: \n",
    "        filtered_sentence_all.append(w)    \n",
    "\n",
    "wordnet_lemmatizer = WordNetLemmatizer()\n",
    "lemmatized_all = []\n",
    "\n",
    "for w in filtered_sentence_all:\n",
    "    lemmatized_w = wordnet_lemmatizer.lemmatize(w)\n",
    "    lemmatized_all.append(lemmatized_w)\n",
    "        \n",
    "bigramFinder_all = nltk.collocations.BigramCollocationFinder.from_words(lemmatized_all)\n",
    "bigram_freq_all = bigramFinder_all.ngram_fd.items()\n",
    "bigramFreqTable_all = pd.DataFrame(list(bigram_freq_all), columns=['bigram','freq']).sort_values(by='freq', ascending=False)\n",
    "\n",
    "bigramFreqTable_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Year</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2015</td>\n",
       "      <td>903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2017</td>\n",
       "      <td>867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2013</td>\n",
       "      <td>853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2016</td>\n",
       "      <td>775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014</td>\n",
       "      <td>757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2018</td>\n",
       "      <td>647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2012</td>\n",
       "      <td>560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2010</td>\n",
       "      <td>534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2011</td>\n",
       "      <td>534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2009</td>\n",
       "      <td>529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2007</td>\n",
       "      <td>502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2008</td>\n",
       "      <td>486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2006</td>\n",
       "      <td>416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2005</td>\n",
       "      <td>376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2004</td>\n",
       "      <td>330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2003</td>\n",
       "      <td>269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2002</td>\n",
       "      <td>234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2001</td>\n",
       "      <td>179</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    index  Year\n",
       "0    2015   903\n",
       "1    2017   867\n",
       "2    2013   853\n",
       "3    2016   775\n",
       "4    2014   757\n",
       "5    2018   647\n",
       "6    2012   560\n",
       "7    2010   534\n",
       "8    2011   534\n",
       "9    2009   529\n",
       "10   2007   502\n",
       "11   2008   486\n",
       "12   2006   416\n",
       "13   2005   376\n",
       "14   2004   330\n",
       "15   2003   269\n",
       "16   2002   234\n",
       "17   2001   179"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_year = pd.DataFrame(abstracts_full[\"Year\"].value_counts()).reset_index()\n",
    "n_year\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>2001</th>\n",
       "      <th>2002</th>\n",
       "      <th>2003</th>\n",
       "      <th>2004</th>\n",
       "      <th>2005</th>\n",
       "      <th>2006</th>\n",
       "      <th>2007</th>\n",
       "      <th>2008</th>\n",
       "      <th>2009</th>\n",
       "      <th>2010</th>\n",
       "      <th>2011</th>\n",
       "      <th>2012</th>\n",
       "      <th>2013</th>\n",
       "      <th>2014</th>\n",
       "      <th>2015</th>\n",
       "      <th>2016</th>\n",
       "      <th>2017</th>\n",
       "      <th>2018</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Frequency</th>\n",
       "      <td>179</td>\n",
       "      <td>234</td>\n",
       "      <td>269</td>\n",
       "      <td>330</td>\n",
       "      <td>376</td>\n",
       "      <td>416</td>\n",
       "      <td>502</td>\n",
       "      <td>486</td>\n",
       "      <td>529</td>\n",
       "      <td>534</td>\n",
       "      <td>534</td>\n",
       "      <td>560</td>\n",
       "      <td>853</td>\n",
       "      <td>757</td>\n",
       "      <td>903</td>\n",
       "      <td>775</td>\n",
       "      <td>867</td>\n",
       "      <td>647</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           2001  2002  2003  2004  2005  2006  2007  2008  2009  2010  2011  \\\n",
       "Frequency   179   234   269   330   376   416   502   486   529   534   534   \n",
       "\n",
       "           2012  2013  2014  2015  2016  2017  2018  \n",
       "Frequency   560   853   757   903   775   867   647  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#itens per year\n",
    "n_year.columns = [\"Year\", \"Frequency\"]\n",
    "n_year = n_year.sort_values(by=\"Year\")\n",
    "n_year_ed = n_year.set_index(\"Year\")\n",
    "n_year_transposed = n_year_ed.transpose()\n",
    "n_year_transposed.columns = [\"2001\", \"2002\", \"2003\", \"2004\", \"2005\", \"2006\", \"2007\", \"2008\", \"2009\", \"2010\", \"2011\", \"2012\", \"2013\", \"2014\", \"2015\", \"2016\", \"2017\", \"2018\"]\n",
    "n_year_transposed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "bigramFreqTable_2001 = pd.read_table('2001_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2002 = pd.read_table('2002_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2003 = pd.read_table('2003_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2004 = pd.read_table('2004_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2005 = pd.read_table('2005_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2006 = pd.read_table('2006_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2007 = pd.read_table('2007_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2008 = pd.read_table('2008_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2009 = pd.read_table('2009_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2010 = pd.read_table('2010_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2011 = pd.read_table('2011_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2012 = pd.read_table('2012_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2013 = pd.read_table('2013_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2014 = pd.read_table('2014_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2015 = pd.read_table('2015_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2016 = pd.read_table('2016_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2017 = pd.read_table('2017_bigram_freq.csv', sep=',', index_col=0)\n",
    "bigramFreqTable_2018 = pd.read_table('2018_bigram_freq.csv', sep=',', index_col=0)\n",
    "\n",
    "dfs = [bigramFreqTable_2001, bigramFreqTable_2002, bigramFreqTable_2003, bigramFreqTable_2004, bigramFreqTable_2005, bigramFreqTable_2006, bigramFreqTable_2007, bigramFreqTable_2008, bigramFreqTable_2009, bigramFreqTable_2010, bigramFreqTable_2011, bigramFreqTable_2012, bigramFreqTable_2013, bigramFreqTable_2014, bigramFreqTable_2015, bigramFreqTable_2016, bigramFreqTable_2017, bigramFreqTable_2018]\n",
    "from functools import reduce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_final = reduce(lambda left,right: pd.merge(left,right,on='bigram'), dfs)\n",
    "df_final.columns = [\"bigram\", \"2001\", \"2002\", \"2003\", \"2004\", \"2005\", \"2006\", \"2007\", \"2008\", \"2009\", \"2010\", \"2011\", \"2012\", \"2013\", \"2014\", \"2015\", \"2016\", \"2017\", \"2018\" ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_final = df_final.set_index(\"bigram\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>2001</th>\n",
       "      <th>2002</th>\n",
       "      <th>2003</th>\n",
       "      <th>2004</th>\n",
       "      <th>2005</th>\n",
       "      <th>2006</th>\n",
       "      <th>2007</th>\n",
       "      <th>2008</th>\n",
       "      <th>2009</th>\n",
       "      <th>2010</th>\n",
       "      <th>2011</th>\n",
       "      <th>2012</th>\n",
       "      <th>2013</th>\n",
       "      <th>2014</th>\n",
       "      <th>2015</th>\n",
       "      <th>2016</th>\n",
       "      <th>2017</th>\n",
       "      <th>2018</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bigram</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>('type', 'strain')</th>\n",
       "      <td>195</td>\n",
       "      <td>278</td>\n",
       "      <td>283</td>\n",
       "      <td>401</td>\n",
       "      <td>481</td>\n",
       "      <td>573</td>\n",
       "      <td>703</td>\n",
       "      <td>743</td>\n",
       "      <td>830</td>\n",
       "      <td>796</td>\n",
       "      <td>744</td>\n",
       "      <td>813</td>\n",
       "      <td>1280</td>\n",
       "      <td>1182</td>\n",
       "      <td>1336</td>\n",
       "      <td>1100</td>\n",
       "      <td>1208</td>\n",
       "      <td>875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>('sp', 'nov')</th>\n",
       "      <td>175</td>\n",
       "      <td>235</td>\n",
       "      <td>253</td>\n",
       "      <td>382</td>\n",
       "      <td>426</td>\n",
       "      <td>482</td>\n",
       "      <td>576</td>\n",
       "      <td>545</td>\n",
       "      <td>614</td>\n",
       "      <td>584</td>\n",
       "      <td>575</td>\n",
       "      <td>614</td>\n",
       "      <td>896</td>\n",
       "      <td>800</td>\n",
       "      <td>939</td>\n",
       "      <td>815</td>\n",
       "      <td>900</td>\n",
       "      <td>660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>('16', 'rrna')</th>\n",
       "      <td>93</td>\n",
       "      <td>135</td>\n",
       "      <td>153</td>\n",
       "      <td>324</td>\n",
       "      <td>495</td>\n",
       "      <td>598</td>\n",
       "      <td>730</td>\n",
       "      <td>719</td>\n",
       "      <td>761</td>\n",
       "      <td>785</td>\n",
       "      <td>799</td>\n",
       "      <td>795</td>\n",
       "      <td>1214</td>\n",
       "      <td>1097</td>\n",
       "      <td>1246</td>\n",
       "      <td>1090</td>\n",
       "      <td>1187</td>\n",
       "      <td>861</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    2001  2002  2003  2004  2005  2006  2007  2008  2009  \\\n",
       "bigram                                                                     \n",
       "('type', 'strain')   195   278   283   401   481   573   703   743   830   \n",
       "('sp', 'nov')        175   235   253   382   426   482   576   545   614   \n",
       "('16', 'rrna')        93   135   153   324   495   598   730   719   761   \n",
       "\n",
       "                    2010  2011  2012  2013  2014  2015  2016  2017  2018  \n",
       "bigram                                                                    \n",
       "('type', 'strain')   796   744   813  1280  1182  1336  1100  1208   875  \n",
       "('sp', 'nov')        584   575   614   896   800   939   815   900   660  \n",
       "('16', 'rrna')       785   799   795  1214  1097  1246  1090  1187   861  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_final.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "relative = df_final.div(n_year_transposed.iloc[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>bigram</th>\n",
       "      <th>('type', 'strain')</th>\n",
       "      <th>('sp', 'nov')</th>\n",
       "      <th>('16', 'rrna')</th>\n",
       "      <th>('new', 'specie')</th>\n",
       "      <th>('degree', 'c')</th>\n",
       "      <th>('fatty', 'acid')</th>\n",
       "      <th>('gc', 'content')</th>\n",
       "      <th>('rrna', 'gene')</th>\n",
       "      <th>('nov', 'type')</th>\n",
       "      <th>('dna-dna', 'hybridization')</th>\n",
       "      <th>...</th>\n",
       "      <th>('chemotaxonomic', 'characterization')</th>\n",
       "      <th>('considered', 'represent')</th>\n",
       "      <th>('also', 'supported')</th>\n",
       "      <th>('profile', 'contained')</th>\n",
       "      <th>('short', 'rod')</th>\n",
       "      <th>('transcribed', 'spacer')</th>\n",
       "      <th>('similarity', 'dna-dna')</th>\n",
       "      <th>('strain', 'contained')</th>\n",
       "      <th>('ubiquinone', '10')</th>\n",
       "      <th>('phylogenetic', 'relative')</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2001</th>\n",
       "      <td>1.089385</td>\n",
       "      <td>0.977654</td>\n",
       "      <td>0.519553</td>\n",
       "      <td>0.508380</td>\n",
       "      <td>0.424581</td>\n",
       "      <td>0.407821</td>\n",
       "      <td>0.391061</td>\n",
       "      <td>0.363128</td>\n",
       "      <td>0.363128</td>\n",
       "      <td>0.357542</td>\n",
       "      <td>...</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.005587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2002</th>\n",
       "      <td>1.188034</td>\n",
       "      <td>1.004274</td>\n",
       "      <td>0.576923</td>\n",
       "      <td>0.209402</td>\n",
       "      <td>0.529915</td>\n",
       "      <td>0.538462</td>\n",
       "      <td>0.384615</td>\n",
       "      <td>0.457265</td>\n",
       "      <td>0.363248</td>\n",
       "      <td>0.354701</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004274</td>\n",
       "      <td>0.008547</td>\n",
       "      <td>0.021368</td>\n",
       "      <td>0.004274</td>\n",
       "      <td>0.008547</td>\n",
       "      <td>0.017094</td>\n",
       "      <td>0.012821</td>\n",
       "      <td>0.025641</td>\n",
       "      <td>0.004274</td>\n",
       "      <td>0.034188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2003</th>\n",
       "      <td>1.052045</td>\n",
       "      <td>0.940520</td>\n",
       "      <td>0.568773</td>\n",
       "      <td>0.189591</td>\n",
       "      <td>0.602230</td>\n",
       "      <td>0.579926</td>\n",
       "      <td>0.237918</td>\n",
       "      <td>0.501859</td>\n",
       "      <td>0.304833</td>\n",
       "      <td>0.308550</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003717</td>\n",
       "      <td>0.003717</td>\n",
       "      <td>0.007435</td>\n",
       "      <td>0.003717</td>\n",
       "      <td>0.033457</td>\n",
       "      <td>0.007435</td>\n",
       "      <td>0.026022</td>\n",
       "      <td>0.018587</td>\n",
       "      <td>0.003717</td>\n",
       "      <td>0.022305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2004</th>\n",
       "      <td>1.215152</td>\n",
       "      <td>1.157576</td>\n",
       "      <td>0.981818</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.412121</td>\n",
       "      <td>0.587879</td>\n",
       "      <td>0.375758</td>\n",
       "      <td>0.909091</td>\n",
       "      <td>0.351515</td>\n",
       "      <td>0.324242</td>\n",
       "      <td>...</td>\n",
       "      <td>0.006061</td>\n",
       "      <td>0.009091</td>\n",
       "      <td>0.009091</td>\n",
       "      <td>0.003030</td>\n",
       "      <td>0.015152</td>\n",
       "      <td>0.021212</td>\n",
       "      <td>0.018182</td>\n",
       "      <td>0.009091</td>\n",
       "      <td>0.003030</td>\n",
       "      <td>0.021212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2005</th>\n",
       "      <td>1.279255</td>\n",
       "      <td>1.132979</td>\n",
       "      <td>1.316489</td>\n",
       "      <td>0.106383</td>\n",
       "      <td>0.465426</td>\n",
       "      <td>0.712766</td>\n",
       "      <td>0.494681</td>\n",
       "      <td>1.263298</td>\n",
       "      <td>0.369681</td>\n",
       "      <td>0.356383</td>\n",
       "      <td>...</td>\n",
       "      <td>0.005319</td>\n",
       "      <td>0.029255</td>\n",
       "      <td>0.007979</td>\n",
       "      <td>0.005319</td>\n",
       "      <td>0.021277</td>\n",
       "      <td>0.013298</td>\n",
       "      <td>0.021277</td>\n",
       "      <td>0.015957</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.034574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2006</th>\n",
       "      <td>1.377404</td>\n",
       "      <td>1.158654</td>\n",
       "      <td>1.437500</td>\n",
       "      <td>0.069712</td>\n",
       "      <td>0.473558</td>\n",
       "      <td>0.694712</td>\n",
       "      <td>0.533654</td>\n",
       "      <td>1.389423</td>\n",
       "      <td>0.382212</td>\n",
       "      <td>0.336538</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002404</td>\n",
       "      <td>0.031250</td>\n",
       "      <td>0.004808</td>\n",
       "      <td>0.002404</td>\n",
       "      <td>0.019231</td>\n",
       "      <td>0.012019</td>\n",
       "      <td>0.024038</td>\n",
       "      <td>0.016827</td>\n",
       "      <td>0.007212</td>\n",
       "      <td>0.021635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2007</th>\n",
       "      <td>1.400398</td>\n",
       "      <td>1.147410</td>\n",
       "      <td>1.454183</td>\n",
       "      <td>0.053785</td>\n",
       "      <td>0.527888</td>\n",
       "      <td>0.846614</td>\n",
       "      <td>0.643426</td>\n",
       "      <td>1.392430</td>\n",
       "      <td>0.256972</td>\n",
       "      <td>0.294821</td>\n",
       "      <td>...</td>\n",
       "      <td>0.005976</td>\n",
       "      <td>0.055777</td>\n",
       "      <td>0.003984</td>\n",
       "      <td>0.003984</td>\n",
       "      <td>0.023904</td>\n",
       "      <td>0.003984</td>\n",
       "      <td>0.015936</td>\n",
       "      <td>0.025896</td>\n",
       "      <td>0.005976</td>\n",
       "      <td>0.027888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2008</th>\n",
       "      <td>1.528807</td>\n",
       "      <td>1.121399</td>\n",
       "      <td>1.479424</td>\n",
       "      <td>0.028807</td>\n",
       "      <td>0.582305</td>\n",
       "      <td>0.775720</td>\n",
       "      <td>0.635802</td>\n",
       "      <td>1.425926</td>\n",
       "      <td>0.288066</td>\n",
       "      <td>0.327160</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004115</td>\n",
       "      <td>0.109053</td>\n",
       "      <td>0.008230</td>\n",
       "      <td>0.004115</td>\n",
       "      <td>0.039095</td>\n",
       "      <td>0.008230</td>\n",
       "      <td>0.008230</td>\n",
       "      <td>0.010288</td>\n",
       "      <td>0.008230</td>\n",
       "      <td>0.028807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2009</th>\n",
       "      <td>1.568998</td>\n",
       "      <td>1.160681</td>\n",
       "      <td>1.438563</td>\n",
       "      <td>0.034026</td>\n",
       "      <td>0.578450</td>\n",
       "      <td>0.780718</td>\n",
       "      <td>0.631380</td>\n",
       "      <td>1.400756</td>\n",
       "      <td>0.323251</td>\n",
       "      <td>0.332703</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001890</td>\n",
       "      <td>0.130435</td>\n",
       "      <td>0.007561</td>\n",
       "      <td>0.003781</td>\n",
       "      <td>0.013233</td>\n",
       "      <td>0.017013</td>\n",
       "      <td>0.015123</td>\n",
       "      <td>0.026465</td>\n",
       "      <td>0.011342</td>\n",
       "      <td>0.020794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2010</th>\n",
       "      <td>1.490637</td>\n",
       "      <td>1.093633</td>\n",
       "      <td>1.470037</td>\n",
       "      <td>0.041199</td>\n",
       "      <td>0.565543</td>\n",
       "      <td>0.855805</td>\n",
       "      <td>0.653558</td>\n",
       "      <td>1.406367</td>\n",
       "      <td>0.314607</td>\n",
       "      <td>0.277154</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003745</td>\n",
       "      <td>0.149813</td>\n",
       "      <td>0.007491</td>\n",
       "      <td>0.005618</td>\n",
       "      <td>0.026217</td>\n",
       "      <td>0.001873</td>\n",
       "      <td>0.018727</td>\n",
       "      <td>0.029963</td>\n",
       "      <td>0.009363</td>\n",
       "      <td>0.037453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2011</th>\n",
       "      <td>1.393258</td>\n",
       "      <td>1.076779</td>\n",
       "      <td>1.496255</td>\n",
       "      <td>0.046816</td>\n",
       "      <td>0.617978</td>\n",
       "      <td>0.883895</td>\n",
       "      <td>0.677903</td>\n",
       "      <td>1.443820</td>\n",
       "      <td>0.217228</td>\n",
       "      <td>0.303371</td>\n",
       "      <td>...</td>\n",
       "      <td>0.013109</td>\n",
       "      <td>0.104869</td>\n",
       "      <td>0.020599</td>\n",
       "      <td>0.007491</td>\n",
       "      <td>0.020599</td>\n",
       "      <td>0.007491</td>\n",
       "      <td>0.020599</td>\n",
       "      <td>0.033708</td>\n",
       "      <td>0.016854</td>\n",
       "      <td>0.044944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2012</th>\n",
       "      <td>1.451786</td>\n",
       "      <td>1.096429</td>\n",
       "      <td>1.419643</td>\n",
       "      <td>0.046429</td>\n",
       "      <td>0.651786</td>\n",
       "      <td>0.914286</td>\n",
       "      <td>0.694643</td>\n",
       "      <td>1.376786</td>\n",
       "      <td>0.246429</td>\n",
       "      <td>0.258929</td>\n",
       "      <td>...</td>\n",
       "      <td>0.005357</td>\n",
       "      <td>0.178571</td>\n",
       "      <td>0.007143</td>\n",
       "      <td>0.019643</td>\n",
       "      <td>0.028571</td>\n",
       "      <td>0.008929</td>\n",
       "      <td>0.023214</td>\n",
       "      <td>0.032143</td>\n",
       "      <td>0.003571</td>\n",
       "      <td>0.028571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2013</th>\n",
       "      <td>1.500586</td>\n",
       "      <td>1.050410</td>\n",
       "      <td>1.423212</td>\n",
       "      <td>0.029308</td>\n",
       "      <td>0.706917</td>\n",
       "      <td>0.917937</td>\n",
       "      <td>0.715123</td>\n",
       "      <td>1.350528</td>\n",
       "      <td>0.202814</td>\n",
       "      <td>0.322392</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004689</td>\n",
       "      <td>0.125440</td>\n",
       "      <td>0.016413</td>\n",
       "      <td>0.012896</td>\n",
       "      <td>0.024619</td>\n",
       "      <td>0.004689</td>\n",
       "      <td>0.019930</td>\n",
       "      <td>0.031653</td>\n",
       "      <td>0.005862</td>\n",
       "      <td>0.036342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2014</th>\n",
       "      <td>1.561427</td>\n",
       "      <td>1.056803</td>\n",
       "      <td>1.449141</td>\n",
       "      <td>0.033025</td>\n",
       "      <td>0.793923</td>\n",
       "      <td>0.945839</td>\n",
       "      <td>0.760898</td>\n",
       "      <td>1.377807</td>\n",
       "      <td>0.269485</td>\n",
       "      <td>0.266843</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001321</td>\n",
       "      <td>0.182299</td>\n",
       "      <td>0.015852</td>\n",
       "      <td>0.019815</td>\n",
       "      <td>0.015852</td>\n",
       "      <td>0.002642</td>\n",
       "      <td>0.015852</td>\n",
       "      <td>0.036988</td>\n",
       "      <td>0.013210</td>\n",
       "      <td>0.038309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2015</th>\n",
       "      <td>1.479513</td>\n",
       "      <td>1.039867</td>\n",
       "      <td>1.379845</td>\n",
       "      <td>0.036545</td>\n",
       "      <td>0.740864</td>\n",
       "      <td>0.940199</td>\n",
       "      <td>0.720930</td>\n",
       "      <td>1.312292</td>\n",
       "      <td>0.198228</td>\n",
       "      <td>0.314507</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004430</td>\n",
       "      <td>0.169435</td>\n",
       "      <td>0.008859</td>\n",
       "      <td>0.021041</td>\n",
       "      <td>0.016611</td>\n",
       "      <td>0.005537</td>\n",
       "      <td>0.012182</td>\n",
       "      <td>0.026578</td>\n",
       "      <td>0.014396</td>\n",
       "      <td>0.039867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2016</th>\n",
       "      <td>1.419355</td>\n",
       "      <td>1.051613</td>\n",
       "      <td>1.406452</td>\n",
       "      <td>0.043871</td>\n",
       "      <td>0.776774</td>\n",
       "      <td>0.934194</td>\n",
       "      <td>0.740645</td>\n",
       "      <td>1.343226</td>\n",
       "      <td>0.184516</td>\n",
       "      <td>0.309677</td>\n",
       "      <td>...</td>\n",
       "      <td>0.009032</td>\n",
       "      <td>0.138065</td>\n",
       "      <td>0.006452</td>\n",
       "      <td>0.021935</td>\n",
       "      <td>0.020645</td>\n",
       "      <td>0.007742</td>\n",
       "      <td>0.018065</td>\n",
       "      <td>0.027097</td>\n",
       "      <td>0.009032</td>\n",
       "      <td>0.042581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2017</th>\n",
       "      <td>1.393310</td>\n",
       "      <td>1.038062</td>\n",
       "      <td>1.369089</td>\n",
       "      <td>0.043829</td>\n",
       "      <td>0.843137</td>\n",
       "      <td>0.944637</td>\n",
       "      <td>0.795848</td>\n",
       "      <td>1.313725</td>\n",
       "      <td>0.162630</td>\n",
       "      <td>0.272203</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003460</td>\n",
       "      <td>0.138408</td>\n",
       "      <td>0.012687</td>\n",
       "      <td>0.017301</td>\n",
       "      <td>0.020761</td>\n",
       "      <td>0.005767</td>\n",
       "      <td>0.009227</td>\n",
       "      <td>0.025375</td>\n",
       "      <td>0.012687</td>\n",
       "      <td>0.033449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2018</th>\n",
       "      <td>1.352396</td>\n",
       "      <td>1.020093</td>\n",
       "      <td>1.330757</td>\n",
       "      <td>0.075734</td>\n",
       "      <td>0.865533</td>\n",
       "      <td>0.925811</td>\n",
       "      <td>0.732612</td>\n",
       "      <td>1.251932</td>\n",
       "      <td>0.142195</td>\n",
       "      <td>0.346213</td>\n",
       "      <td>...</td>\n",
       "      <td>0.012365</td>\n",
       "      <td>0.126739</td>\n",
       "      <td>0.012365</td>\n",
       "      <td>0.035549</td>\n",
       "      <td>0.015456</td>\n",
       "      <td>0.003091</td>\n",
       "      <td>0.007728</td>\n",
       "      <td>0.024730</td>\n",
       "      <td>0.009274</td>\n",
       "      <td>0.017002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18 rows × 662 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "bigram  ('type', 'strain')  ('sp', 'nov')  ('16', 'rrna')  ('new', 'specie')  \\\n",
       "2001              1.089385       0.977654        0.519553           0.508380   \n",
       "2002              1.188034       1.004274        0.576923           0.209402   \n",
       "2003              1.052045       0.940520        0.568773           0.189591   \n",
       "2004              1.215152       1.157576        0.981818           0.127273   \n",
       "2005              1.279255       1.132979        1.316489           0.106383   \n",
       "2006              1.377404       1.158654        1.437500           0.069712   \n",
       "2007              1.400398       1.147410        1.454183           0.053785   \n",
       "2008              1.528807       1.121399        1.479424           0.028807   \n",
       "2009              1.568998       1.160681        1.438563           0.034026   \n",
       "2010              1.490637       1.093633        1.470037           0.041199   \n",
       "2011              1.393258       1.076779        1.496255           0.046816   \n",
       "2012              1.451786       1.096429        1.419643           0.046429   \n",
       "2013              1.500586       1.050410        1.423212           0.029308   \n",
       "2014              1.561427       1.056803        1.449141           0.033025   \n",
       "2015              1.479513       1.039867        1.379845           0.036545   \n",
       "2016              1.419355       1.051613        1.406452           0.043871   \n",
       "2017              1.393310       1.038062        1.369089           0.043829   \n",
       "2018              1.352396       1.020093        1.330757           0.075734   \n",
       "\n",
       "bigram  ('degree', 'c')  ('fatty', 'acid')  ('gc', 'content')  \\\n",
       "2001           0.424581           0.407821           0.391061   \n",
       "2002           0.529915           0.538462           0.384615   \n",
       "2003           0.602230           0.579926           0.237918   \n",
       "2004           0.412121           0.587879           0.375758   \n",
       "2005           0.465426           0.712766           0.494681   \n",
       "2006           0.473558           0.694712           0.533654   \n",
       "2007           0.527888           0.846614           0.643426   \n",
       "2008           0.582305           0.775720           0.635802   \n",
       "2009           0.578450           0.780718           0.631380   \n",
       "2010           0.565543           0.855805           0.653558   \n",
       "2011           0.617978           0.883895           0.677903   \n",
       "2012           0.651786           0.914286           0.694643   \n",
       "2013           0.706917           0.917937           0.715123   \n",
       "2014           0.793923           0.945839           0.760898   \n",
       "2015           0.740864           0.940199           0.720930   \n",
       "2016           0.776774           0.934194           0.740645   \n",
       "2017           0.843137           0.944637           0.795848   \n",
       "2018           0.865533           0.925811           0.732612   \n",
       "\n",
       "bigram  ('rrna', 'gene')  ('nov', 'type')  ('dna-dna', 'hybridization')  \\\n",
       "2001            0.363128         0.363128                      0.357542   \n",
       "2002            0.457265         0.363248                      0.354701   \n",
       "2003            0.501859         0.304833                      0.308550   \n",
       "2004            0.909091         0.351515                      0.324242   \n",
       "2005            1.263298         0.369681                      0.356383   \n",
       "2006            1.389423         0.382212                      0.336538   \n",
       "2007            1.392430         0.256972                      0.294821   \n",
       "2008            1.425926         0.288066                      0.327160   \n",
       "2009            1.400756         0.323251                      0.332703   \n",
       "2010            1.406367         0.314607                      0.277154   \n",
       "2011            1.443820         0.217228                      0.303371   \n",
       "2012            1.376786         0.246429                      0.258929   \n",
       "2013            1.350528         0.202814                      0.322392   \n",
       "2014            1.377807         0.269485                      0.266843   \n",
       "2015            1.312292         0.198228                      0.314507   \n",
       "2016            1.343226         0.184516                      0.309677   \n",
       "2017            1.313725         0.162630                      0.272203   \n",
       "2018            1.251932         0.142195                      0.346213   \n",
       "\n",
       "bigram              ...               ('chemotaxonomic', 'characterization')  \\\n",
       "2001                ...                                             0.005587   \n",
       "2002                ...                                             0.004274   \n",
       "2003                ...                                             0.003717   \n",
       "2004                ...                                             0.006061   \n",
       "2005                ...                                             0.005319   \n",
       "2006                ...                                             0.002404   \n",
       "2007                ...                                             0.005976   \n",
       "2008                ...                                             0.004115   \n",
       "2009                ...                                             0.001890   \n",
       "2010                ...                                             0.003745   \n",
       "2011                ...                                             0.013109   \n",
       "2012                ...                                             0.005357   \n",
       "2013                ...                                             0.004689   \n",
       "2014                ...                                             0.001321   \n",
       "2015                ...                                             0.004430   \n",
       "2016                ...                                             0.009032   \n",
       "2017                ...                                             0.003460   \n",
       "2018                ...                                             0.012365   \n",
       "\n",
       "bigram  ('considered', 'represent')  ('also', 'supported')  \\\n",
       "2001                       0.005587               0.005587   \n",
       "2002                       0.008547               0.021368   \n",
       "2003                       0.003717               0.007435   \n",
       "2004                       0.009091               0.009091   \n",
       "2005                       0.029255               0.007979   \n",
       "2006                       0.031250               0.004808   \n",
       "2007                       0.055777               0.003984   \n",
       "2008                       0.109053               0.008230   \n",
       "2009                       0.130435               0.007561   \n",
       "2010                       0.149813               0.007491   \n",
       "2011                       0.104869               0.020599   \n",
       "2012                       0.178571               0.007143   \n",
       "2013                       0.125440               0.016413   \n",
       "2014                       0.182299               0.015852   \n",
       "2015                       0.169435               0.008859   \n",
       "2016                       0.138065               0.006452   \n",
       "2017                       0.138408               0.012687   \n",
       "2018                       0.126739               0.012365   \n",
       "\n",
       "bigram  ('profile', 'contained')  ('short', 'rod')  ('transcribed', 'spacer')  \\\n",
       "2001                    0.005587          0.005587                   0.005587   \n",
       "2002                    0.004274          0.008547                   0.017094   \n",
       "2003                    0.003717          0.033457                   0.007435   \n",
       "2004                    0.003030          0.015152                   0.021212   \n",
       "2005                    0.005319          0.021277                   0.013298   \n",
       "2006                    0.002404          0.019231                   0.012019   \n",
       "2007                    0.003984          0.023904                   0.003984   \n",
       "2008                    0.004115          0.039095                   0.008230   \n",
       "2009                    0.003781          0.013233                   0.017013   \n",
       "2010                    0.005618          0.026217                   0.001873   \n",
       "2011                    0.007491          0.020599                   0.007491   \n",
       "2012                    0.019643          0.028571                   0.008929   \n",
       "2013                    0.012896          0.024619                   0.004689   \n",
       "2014                    0.019815          0.015852                   0.002642   \n",
       "2015                    0.021041          0.016611                   0.005537   \n",
       "2016                    0.021935          0.020645                   0.007742   \n",
       "2017                    0.017301          0.020761                   0.005767   \n",
       "2018                    0.035549          0.015456                   0.003091   \n",
       "\n",
       "bigram  ('similarity', 'dna-dna')  ('strain', 'contained')  \\\n",
       "2001                     0.005587                 0.005587   \n",
       "2002                     0.012821                 0.025641   \n",
       "2003                     0.026022                 0.018587   \n",
       "2004                     0.018182                 0.009091   \n",
       "2005                     0.021277                 0.015957   \n",
       "2006                     0.024038                 0.016827   \n",
       "2007                     0.015936                 0.025896   \n",
       "2008                     0.008230                 0.010288   \n",
       "2009                     0.015123                 0.026465   \n",
       "2010                     0.018727                 0.029963   \n",
       "2011                     0.020599                 0.033708   \n",
       "2012                     0.023214                 0.032143   \n",
       "2013                     0.019930                 0.031653   \n",
       "2014                     0.015852                 0.036988   \n",
       "2015                     0.012182                 0.026578   \n",
       "2016                     0.018065                 0.027097   \n",
       "2017                     0.009227                 0.025375   \n",
       "2018                     0.007728                 0.024730   \n",
       "\n",
       "bigram  ('ubiquinone', '10')  ('phylogenetic', 'relative')  \n",
       "2001                0.005587                      0.005587  \n",
       "2002                0.004274                      0.034188  \n",
       "2003                0.003717                      0.022305  \n",
       "2004                0.003030                      0.021212  \n",
       "2005                0.002660                      0.034574  \n",
       "2006                0.007212                      0.021635  \n",
       "2007                0.005976                      0.027888  \n",
       "2008                0.008230                      0.028807  \n",
       "2009                0.011342                      0.020794  \n",
       "2010                0.009363                      0.037453  \n",
       "2011                0.016854                      0.044944  \n",
       "2012                0.003571                      0.028571  \n",
       "2013                0.005862                      0.036342  \n",
       "2014                0.013210                      0.038309  \n",
       "2015                0.014396                      0.039867  \n",
       "2016                0.009032                      0.042581  \n",
       "2017                0.012687                      0.033449  \n",
       "2018                0.009274                      0.017002  \n",
       "\n",
       "[18 rows x 662 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transposed_relative = relative.T\n",
    "transposed_relative"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['year', 'freq'], dtype='object')"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_specie = transposed_relative[\"('sp', 'nov')\"]\n",
    "new_specie.index.name = 'year'\n",
    "new_specie = new_specie.reset_index()\n",
    "new_specie.columns = [\"year\", \"freq\"]\n",
    "new_specie.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['year', 'freq'], dtype='object')"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_genus = transposed_relative[\"('gen', 'nov')\"]\n",
    "new_genus.index.name = 'year'\n",
    "new_genus = new_genus.reset_index()\n",
    "new_genus.columns = [\"year\", \"freq\"]\n",
    "new_genus.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['year', 'freq'], dtype='object')"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s16_rrna = transposed_relative[\"('16', 'rrna')\"]\n",
    "s16_rrna.index.name = 'year'\n",
    "s16_rrna = s16_rrna.reset_index()\n",
    "s16_rrna.columns = [\"year\", \"freq\"]\n",
    "s16_rrna.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['year', 'freq'], dtype='object')"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fatty_acid = transposed_relative[\"('fatty', 'acid')\"]\n",
    "fatty_acid.index.name = 'year'\n",
    "fatty_acid = fatty_acid.reset_index()\n",
    "fatty_acid.columns = [\"year\", \"freq\"]\n",
    "fatty_acid.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['year', 'freq'], dtype='object')"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gc_content = transposed_relative[\"('gc', 'content')\"]\n",
    "gc_content.index.name = 'year'\n",
    "gc_content = gc_content.reset_index()\n",
    "gc_content.columns = [\"year\", \"freq\"]\n",
    "gc_content.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['year', 'freq'], dtype='object')"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phylogenetic_analysis = transposed_relative[\"('phylogenetic', 'analysis')\"]\n",
    "phylogenetic_analysis.index.name = 'year'\n",
    "phylogenetic_analysis = phylogenetic_analysis.reset_index()\n",
    "phylogenetic_analysis.columns = [\"year\", \"freq\"]\n",
    "phylogenetic_analysis.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['year', 'freq'], dtype='object')"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "polar_lipid = transposed_relative[\"('polar', 'lipid')\"]\n",
    "polar_lipid.index.name = 'year'\n",
    "polar_lipid = polar_lipid.reset_index()\n",
    "polar_lipid.columns = [\"year\", \"freq\"]\n",
    "polar_lipid.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['year', 'freq'], dtype='object')"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dna_hybridization = transposed_relative[\"('dna-dna', 'hybridization')\"]\n",
    "dna_hybridization.index.name = 'year'\n",
    "dna_hybridization = dna_hybridization.reset_index()\n",
    "dna_hybridization.columns = [\"year\", \"freq\"]\n",
    "dna_hybridization.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['year', 'freq'], dtype='object')"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "respiratory_quinone = transposed_relative[\"('respiratory', 'quinone')\"]\n",
    "respiratory_quinone.index.name = 'year'\n",
    "respiratory_quinone = respiratory_quinone.reset_index()\n",
    "respiratory_quinone.columns = [\"year\", \"freq\"]\n",
    "respiratory_quinone.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Figure 8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "import matplotlib.pyplot as plt\n",
    "from matplotlib import rcParams\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(12,8))\n",
    "\n",
    "for axis in ['top','bottom','left','right']:\n",
    "    ax.spines[axis].set_linewidth(1.5)\n",
    "\n",
    "ax.xaxis.set_tick_params(width=1.5, labelsize=14)\n",
    "ax.yaxis.set_tick_params(width=1.5, labelsize=14)\n",
    "#pal = sns.color_palette(\"Set1\")\n",
    "rcParams['font.family'] = 'sans-serif'\n",
    "rcParams['font.sans-serif'] = ['Arial']\n",
    "\n",
    "sns.pointplot(x=\"year\", y=\"freq\", data=s16_rrna, color=\"#f77189\")\n",
    "fig.text(x = 0.91, y = 0.76, s = \"16 rrna\", family='sans-serif', fontsize=14, weight = \"bold\", color=\"#f77189\")\n",
    "sns.pointplot(x=\"year\", y=\"freq\", data=fatty_acid, color=\"#bb9832\")\n",
    "fig.text(x = 0.91, y = 0.57, s = \"fatty acid\", family='sans-serif', fontsize=14, weight = \"bold\", color=\"#bb9832\")\n",
    "sns.pointplot(x=\"year\", y=\"freq\", data=gc_content, color=\"#50b131\")\n",
    "fig.text(x = 0.91, y = 0.48, s = \"gc content\", family='sans-serif', fontsize=14, weight = \"bold\", color=\"#50b131\")\n",
    "sns.pointplot(x=\"year\", y=\"freq\", data=phylogenetic_analysis, color=\"#36ada4\")\n",
    "fig.text(x = 0.91, y = 0.42, s = \"phylogenetic analysis\", family='sans-serif', fontsize=14, weight = \"bold\", color=\"#36ada4\")\n",
    "sns.pointplot(x=\"year\", y=\"freq\", data=polar_lipid, color=\"#3ba3ec\")\n",
    "fig.text(x = 0.91, y = 0.50, s = \"polar lipid\", family='sans-serif', fontsize=14, weight = \"bold\", color=\"#3ba3ec\")\n",
    "sns.pointplot(x=\"year\", y=\"freq\", data=dna_hybridization, color=\"#e866f4\")\n",
    "fig.text(x = 0.91, y = 0.32, s = \"dna-dna hybridization\", family='sans-serif', fontsize=14, weight = \"bold\", color=\"#e866f4\")\n",
    "sns.pointplot(x=\"year\", y=\"freq\", data=respiratory_quinone, color=\"#a48cf4\")\n",
    "fig.text(x = 0.91, y = 0.30, s = \"respiratory quinone\", weight = \"bold\", family='sans-serif', fontsize=14, color=\"#a48cf4\")\n",
    "\n",
    "fig.savefig('bigram.svg', format='svg', dpi=600)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
